### ISSUE TITLE
[Obsolete] not reported for op_True and op_False

### ISSUE THREAD

**Issue by amcasey** [dotnet/roslyn#9306]
``` C#
using System;

namespace ConsoleApplication251
{
    class Program
    {
        static void Main(string[] args)
        {
            Program p1 = new Program();
            Program p2 = new Program();
            Console.Write(p1 && p2);
            Console.Write(p1 || p2);
            if (p1) { }
        }

        [Obsolete("true")]
        public static bool operator true(Program p) => true;
        [Obsolete("false")]
        public static bool operator false(Program p) => false;
        public static Program operator &(Program p1, Program p2) => p1;
        public static Program operator |(Program p1, Program p2) => p2;
    }
}
```

`&&` should report that `op_False` is obsolete (as in dev12)
`||` should report that `op_True` is obsolete (as in dev12)
`if` correctly reports that `op_True` is obsolete.

**Comment by amcasey** [comment]
FYI @jaredpar for triaging.

**Comment by amcasey** [comment]
FYI @genlu, who ran into this issue.

**Comment by jaredpar** [comment]
Fixed now with our recent obsolete checking work.
